types/web3/eth/types.d.ts

```diff
@@ -82,7 +82,7 @@ export interface Transaction {
 export interface TransactionObject<T> {
 	arguments: any[];
 	call(tx?: Tx): Promise<T>;
-	send(tx?: Tx): PromiEvent<T>;
+	send(tx?: Tx): PromiEvent<TransactionReceipt>;
```

Comment:
If I read [this](https://web3js.readthedocs.io/en/1.0/web3-eth-contract.html#id22) correctly the `PromiEvent` (which is not a `Promise`!!) can return either a receipt or a contract instance. It will resolve with a contract instance when called `send` is called on the return value from `contract.deploy`. However it will resolve with a `TransactionReceipt` when called in (I think) any other case... So always `TransactionReceipt` is incorrect.